casts/bars/src/UILargeTitleHeader/LargeTitleHeaderContent.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { StyleSheet, View } from 'react-native';
+import Animated, {
+    Extrapolate,
+    interpolate,
+    useAnimatedStyle,
+    useDerivedValue,
+    useSharedValue,
+} from 'react-native-reanimated';
+
+import { TouchableOpacity } from '@tonlabs/uikit.controls';
+import { UILabel, UILabelColors, UILabelRoles } from '@tonlabs/uikit.themes';
+
+import type { UILargeTitleHeaderProps } from './types';
+import { useRubberBandEffectDistance } from './useRubberBandEffectDistance';
+import { runOnUIPlatformSelect } from './useScrollHandler/runOnUIPlatformSelect';
+import { UIConstant } from '../constants';
+
+const AnimatedUILabel = Animated.createAnimatedComponent(UILabel);
+
+// @inline
+const LARGE_TITLE_SCALE_WEB = 1.1;
+// @inline
+const LARGE_TITLE_SCALE_DEFAULT = 1.2;
+
+function useLargeTitleScale() {
+    return useDerivedValue(() => {
+        return runOnUIPlatformSelect({
+            web: LARGE_TITLE_SCALE_WEB,
+            default: LARGE_TITLE_SCALE_DEFAULT,
+        });
+    });
+}
+
+export const LargeTitleHeaderContent = React.memo(function LargeTitleHeaderContent({
+    label,
+    labelTestID,
+    title,
+    titleTestID,
+    note,
+    noteTestID,
+    onHeaderLargeTitlePress,
+    onHeaderLargeTitleLongPress,
+    shift,
+}: Pick<
+    UILargeTitleHeaderProps,
+    | 'label'
+    | 'labelTestID'
+    | 'title'
+    | 'titleTestID'
+    | 'note'
+    | 'noteTestID'
+    | 'onHeaderLargeTitlePress'
+    | 'onHeaderLargeTitleLongPress'
+> & {
+    shift: Animated.SharedValue<number>;
+}) {
+    const titleWidth = useSharedValue(0);
+    const rubberBandEffectDistance = useRubberBandEffectDistance();
+    const largeTitleScale = useLargeTitleScale();
+    const largeTitleStyle = useAnimatedStyle(() => {
+        return {
+            transform: [
+                {
+                    scale: interpolate(
+                        shift.value,
+                        [0, rubberBandEffectDistance.value],
+                        [1, largeTitleScale.value],
+                        {
+                            extrapolateLeft: Extrapolate.CLAMP,
+                        },
+                    ),
+                },
+                {
+                    translateX: interpolate(
```

Comment:
I wonder if we can "compensate" the scale in some other way without touching `translateX`...

Reply:
Maybe it's possible to use raw transformation matrix somehow, it's just I don't know how to use them yet 😅

Reply:
Also, it might be something connected to the view itself aligned to "center"... at least it would make sense

Reply:
No, it's actually because scale is applied to a center of a view, there're APIs to specify scale origin (`transform-origin` on web), but it doesn't exist in RN